data-prepper-plugins/s3-source/src/main/java/com/amazon/dataprepper/plugins/source/S3Source.java

```diff
@@ -12,52 +12,43 @@
 import com.amazon.dataprepper.model.event.Event;
 import com.amazon.dataprepper.model.record.Record;
 import com.amazon.dataprepper.model.source.Source;
-import org.apache.commons.lang3.NotImplementedException;
-import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
-import software.amazon.awssdk.regions.Region;
-import software.amazon.awssdk.services.s3.S3Client;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.services.sqs.SqsClient;
 import software.amazon.awssdk.services.sts.StsClient;
 
-import java.util.Objects;
-
 @DataPrepperPlugin(name = "s3", pluginType = Source.class, pluginConfigurationType = S3SourceConfig.class)
 public class S3Source implements Source<Record<Event>> {
+    private static final Logger LOG = LoggerFactory.getLogger(S3Source.class);
 
     private final PluginMetrics pluginMetrics;
     private final S3SourceConfig s3SourceConfig;
+    private software.amazon.awssdk.services.s3.S3Client s3Client;
+    private SqsClient sqsClient;
+    private Thread sqsWorkerThread;
 
     @DataPrepperPluginConstructor
     public S3Source(PluginMetrics pluginMetrics, final S3SourceConfig s3SourceConfig) {
         this.pluginMetrics = pluginMetrics;
         this.s3SourceConfig = s3SourceConfig;
-
-        awsCredentialsProvider = createCredentialsProvider();
-        s3Client = createS3Client(awsCredentialsProvider);
-
-        throw new NotImplementedException();
     }
 
     @Override
     public void start(Buffer<Record<Event>> buffer) {
+        if (buffer == null) {
+            throw new IllegalStateException("Buffer provided is null");
+        }
 
+        LOG.info("Creating SQS and S3 client");
+        StsClient stsClient = StsClient.create();
+        this.s3Client = new S3ClientAuthentication(s3SourceConfig).createS3Client(stsClient);
+        this.sqsClient = new SqsClientAuthentication(s3SourceConfig).createSqsClient(stsClient);
+
+        sqsWorkerThread = new Thread(new SqsWorker(sqsClient, s3Client, s3SourceConfig));
     }
 
     @Override
     public void stop() {
-
-    }
-
-    private final AwsCredentialsProvider awsCredentialsProvider;
-    private final S3Client s3Client;
-
-    private AwsCredentialsProvider createCredentialsProvider() {
-        return Objects.requireNonNull(s3SourceConfig.getAWSAuthentication().authenticateAwsConfiguration(StsClient.create()));
-    }
-
-    private S3Client createS3Client(final AwsCredentialsProvider awsCredentialsProvider) {
-        return S3Client.builder()
-                .region(Region.of(s3SourceConfig.getAWSAuthentication().getAwsRegion()))
-                .credentialsProvider(awsCredentialsProvider)
-                .build();
+        Thread.currentThread().interrupt();
```

Comment:
Data Prepper core is calling the `stop()` method. Thus, this current code is going to interrupt the Data Prepper core thread.

Also, rather than use `interrupt()`, I'd like to have a `running`/`shutdown` boolean flag in the worker thread. You then can use `while(running) { ... }` instead of `while(true) { ... }`.